extensions/eclipse-che-theia-workspace/src/browser/che-quick-open-workspace.ts

```diff
@@ -0,0 +1,229 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { inject, injectable } from 'inversify';
+import {
+    QuickOpenGroupItem,
+    QuickOpenItem,
+    QuickOpenMode,
+    QuickOpenModel
+} from '@theia/core/lib/common/quick-open-model';
+import { LabelProvider, QuickOpenService } from '@theia/core/lib/browser';
+import { CommandContribution, CommandRegistry } from '@theia/core/lib/common';
+import { WorkspaceCommands } from '@theia/workspace/lib/browser/workspace-commands';
+import { CheApiService } from '@eclipse-che/theia-plugin-ext/lib/common/che-protocol';
+import { che as cheApi } from '@eclipse-che/api';
+import { OauthUtils } from '@eclipse-che/theia-plugin-ext/lib/browser/oauth-utils';
+import { AbstractDialog } from '@theia/core/lib/browser/dialogs';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import * as moment from 'moment';
+import { Message } from '@theia/core/lib/browser/widgets/index';
+import { Key } from '@theia/core/lib/browser/keyboard/keys';
+
+@injectable()
+export class QuickOpenCheWorkspace implements QuickOpenModel, CommandContribution {
+    protected items: QuickOpenGroupItem[];
+    protected currentWorkspace: cheApi.workspace.Workspace;
+
+    @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService;
+    @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry;
+    @inject(CheApiService) protected readonly cheApi: CheApiService;
+    @inject(OauthUtils) protected readonly oAuthUtils: OauthUtils;
+    @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
+    @inject(MessageService) protected readonly messageService: MessageService;
+
+    registerCommands(commands: CommandRegistry): void {
+        // VS Code command id
+        commands.unregisterCommand('workbench.action.openRecent');
```

Comment:
If I've followed the conversation on the issue correctly, eventually we'll end up with two sets of commands ("Open Workspace") and "Open \<whatever we rename che workspaces to\>". Why not just leave the current commands in place and have call the Che workspace "Che workspace"?
Currently, the ability to "Open Workspace" and "Add Folder to Workspace" play an important role when doing development of che-theia. Please don't take them away.

Reply:
>we'll end up with two sets of commands ("Open Workspace") and "Open <whatever we rename che workspaces to>"

That is true. According to https://github.com/eclipse/che/issues/17106#issue-631935000 and https://github.com/eclipse/che/issues/17034 the current command will be replaced with Che's one. And with next PR (for the issue: https://github.com/eclipse/che/issues/17106#issue-631935000) default Theia's command will be renamed to a new one. But this is out of scope of current PR. This is the first step.